crates/runtime/src/http/routes.rs

```diff
@@ -74,13 +72,26 @@ pub(crate) fn routes(
             .layer(Extension(Arc::clone(&rt.embeds)));
     }
 
-    router = router
+    authenticated_router = authenticated_router
         .layer(Extension(Arc::clone(&rt.app)))
         .layer(Extension(Arc::clone(&rt.df)))
         .layer(Extension(Arc::clone(rt)))
         .layer(Extension(rt.metrics_endpoint))
         .layer(Extension(config));
-    router
+
+    // If we have an auth layer, add it to the authenticated router
+    if let Some(auth_layer) = auth_layer {
+        authenticated_router = authenticated_router.route_layer(auth_layer);
```

Comment:
We use `route_layer` here instead of `layer` because we only want this to run on defined routes. Otherwise it would run on routes that didn't exist and potentially return a `401 Unauthorized` instead of a `404 Not Found`.